pkg/client/unversioned/helper.go

```diff
@@ -403,12 +400,12 @@ func RESTClientFor(config *Config) (*RESTClient, error) {
 		return nil, fmt.Errorf("Codec is required when initializing a RESTClient")
 	}
 
-	baseURL, err := defaultServerUrlFor(config)
+	baseURL, prefix, err := defaultServerUrlFor(config)
 	if err != nil {
 		return nil, err
 	}
 
-	client := NewRESTClient(baseURL, config.Version, config.Codec, config.QPS, config.Burst)
+	client := NewRESTClient(baseURL, prefix, config.Version, config.Codec, config.QPS, config.Burst)
```

Comment:
It's my understanding that `Config.Prefix` may point to `.../api` or `.../apis` but not `.../healthz`.

`baseURL` is not a good name. It may actually contain the path to the api version, e.g. `/api/v1`. Probably more aptly named `unversionedBase`.